### PR TITLE
Add a VxLAN resource pool.

### PR DESCRIPTION
**Pull request by dpoltavets**
Part of: #647.

### CODE REVIEW

File: services/src/kilda-core/kilda-persistence-neo4j/src/test/java/org/openkilda/persistence/repositories/impl/Neo4jVxlanRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                .build();
+        vxlanRepository.createOrUpdate(vxlan);
+
+        vxlanRepository.delete(vxlan);
+
+        assertEquals(0, vxlanRepository.findAll().size());
+    }
+
+    @Test
+    public void shouldDeleteFoundFlowMeter() {
```
Comment (timofei-durakov) [inline services/src/kilda-core/kilda-persistence-neo4j/src/test/java/org/openkilda/persistence/repositories/impl/Neo4jVxlanRepositoryTest.java]: wrong name

File: services/src/kilda-core/kilda-persistence-neo4j/src/test/java/org/openkilda/persistence/repositories/impl/Neo4jVxlanRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+        Collection<Vxlan> allVxlans = vxlanRepository.findAll();
+        Vxlan foundVxlan = allVxlans.iterator().next();
+
+        assertEquals(vxlan.getUnmaskedVxlan(), foundVxlan.getUnmaskedVxlan());
+        assertEquals(TEST_FLOW_ID, foundVxlan.getFlowId());
+    }
+
+    @Test
+    public void shouldDeleteFlowMeter() {
```
Comment (timofei-durakov) [inline services/src/kilda-core/kilda-persistence-neo4j/src/test/java/org/openkilda/persistence/repositories/impl/Neo4jVxlanRepositoryTest.java]: wrong name

File: services/src/kilda-core/kilda-persistence-neo4j/src/main/java/org/openkilda/persistence/repositories/impl/Neo4jVxlanRepository.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+        return loadAll(pathIdFilter);
+    }
+
+    @Override
+    public Optional<Integer> findUnassignedVxlan(int defaultVxlan) {
+        Map<String, Object> parameters = ImmutableMap.of(
+                "default_vxlan", defaultVxlan);
+
+        // The query returns the default_vlan if it's not used in any transit_vlan,
```
Comment (timofei-durakov) [inline services/src/kilda-core/kilda-persistence-neo4j/src/main/java/org/openkilda/persistence/repositories/impl/Neo4jVxlanRepository.java]: it this comment valid for vxlan encap

File: services/wfm/src/test/java/org/openkilda/wfm/share/flow/resources/vxlan/VxlanPoolTest.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    public void setUp() {
+        vxlanPool = new VxlanPool(persistenceManager, 100, 110);
+
+        SwitchRepository switchRepository = persistenceManager.getRepositoryFactory().createSwitchRepository();
+        switchRepository.createOrUpdate(switch1);
+        switchRepository.createOrUpdate(switch2);
+    }
+
+    @Test
+    public void vlanIdPool() {
```
Comment (timofei-durakov) [inline services/wfm/src/test/java/org/openkilda/wfm/share/flow/resources/vxlan/VxlanPoolTest.java]: vxlan

File: services/wfm/src/test/java/org/openkilda/wfm/share/flow/resources/vxlan/VxlanPoolTest.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        assertEquals(101, fourth);
+
+        flow.setFlowId("flow_5");
+        int fifth = vxlanPool.allocate(flow, new PathId("path_5")).getVxlan().getUnmaskedVxlan();
+        assertEquals(103, fifth);
+    }
+
+
+    @Test(expected = ResourceNotAvailableException.class)
+    public void vlanPoolFullTest() {
```
Comment (timofei-durakov) [inline services/wfm/src/test/java/org/openkilda/wfm/share/flow/resources/vxlan/VxlanPoolTest.java]: vxlan

File: services/wfm/src/main/java/org/openkilda/wfm/share/flow/resources/vxlan/VxlanPool.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+import org.openkilda.wfm.share.flow.resources.EncapsulationResourcesProvider;
+import org.openkilda.wfm.share.flow.resources.ResourceNotAvailableException;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * The resource pool is responsible for transit vlan de-/allocation.
```
Comment (sergii-iakovenko) [inline services/wfm/src/main/java/org/openkilda/wfm/share/flow/resources/vxlan/VxlanPool.java]: Shouldn't "transit vlan" be replaced with "vxlan"?

**Pull request merged**